Tests/MongoSwiftTests/AsyncTestUtils.swift

```diff
@@ -14,6 +16,34 @@ extension MongoClient {
                 pemFile: URL(string: MongoSwiftTestCase.sslPEMKeyFilePath ?? "")
             )
         }
-        return try MongoClient(uri, options: opts)
+        return try MongoClient(uri, using: eventLoopGroup, options: opts)
+    }
+
+    internal func syncCloseOrLogError() {
+        do {
+            try self.close().wait()
+        } catch {
+            print("Error closing test client: \(error)")
+        }
+    }
+}
+
+extension MongoSwiftTestCase {
+    internal func withTestClient<T>(options: ClientOptions? = nil, f: (MongoClient) throws -> T) throws -> T {
```

Comment:
eventually this may look more like `withTestNamespace`, but for now we only have a working async client so I left it like this.